policies/templates/gcp_storage_location_v1.yaml

```diff
@@ -0,0 +1,112 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This template is for policies restricting the locations 
+# of BigQuery datasets to specific locations in GCP. It supports 
+# allowlist or denylist modes, as well as exempting selected 
+# datasets from the list.
+
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-storage-location-v1
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPStorageLocationConstraintV1
+        plural: gcpstoragelocationconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties:
+            mode:
+              type: string
+              enum: [denylist, allowlist]
+              description: "String identifying the operational mode, allowlist or denylist. In allowlist mode, 
+              datasets are only allowed in the locations specified in the 'locations' parameter. In denylist mode, 
+              resources are allowed in all locations except those listed in the 'locations' parameter."
+            exemptions:
+              type: array
+              items: string
+              description: "Array of storage buckets to exempt from location restriction. String values in the array should 
+              correspond to the full name values of exempted storage buckets."
+            locations:
+              type: array
+              items: string
+              description: "Array of location names to be allowed or denied. Should be the the location name, whether regional  
+              (e.g. us-west2) or multi-regional (e.g. EU), as defined at https://cloud.google.com/storage/docs/locations."
```

Comment:
what about the new dual regional hotness?

Reply:
updated parameter description